starcheck/src/starcheck.pl

```diff
@@ -108,6 +111,47 @@
     return state
 
 
+def get_now_time():
+    """
+    Return the current time in YYYY:DOY format
+    """
+    return DateTime().date
+
+
+def run_start_time(run_start, backstop_start):
+    """
+    Determine a reasonable reference run start time based on the supplied
+    run start time and the time of the first backstop command.  This
+    code uses a small hack so that a negative number is interpreted
+    as the desired "days back" from the backstop start time.  All other
+    Chandra.Time compatible formats for run start are used as absolute
+    times (which will then be passed to the thermal model code as the
+    time before which telemetry should be found for an initial state.
+
+    :param run_start: supplied run start time in a Chandra.Time format,
+                      negative numbers interpreted as "days back" relative
+                      to first backstop command
+    :param backstop_start: time of first backstop command
+    :returns: YYYY:DOY string of reference run start time
+    """
+
+    run_start = de_bytestr(run_start)
+    ref_time = DateTime(run_start)
+    # For the special case where run_start casts as a float
+    # check to see if it is negative and if so, override the reference
+    # time with a time run_start days back from backstop start
+    try:
+        run_start = float(run_start)
```

Comment:
@taldcroft Is this about the level of "hacky" that you had in mind?

Reply:
Looks fine.  Just some nits:
- The first `ref_time = DateTime(run_start)` should be in the `except` block since it should only run if the input is not a float.
- Do `except TypeError:`
- Put the `if run_start < 0:` stuff into an `else:` block.  This part should not be excepted from errors, you always want to limit exception catching to just the code in question.
- There should probably be an `else:` block on the `if run_start < 0` which raises a ValueError.  The current behavior for positive values will give an unexpected result (i.e. a date near 1998:001) and likely make a difficult-to-understand crash later.

Reply:
Regarding: `ref_time = DateTime(run_start)` should be in the `except` block, I wasn't actually sure about this.  I was OK with a positive float being handled that way with the idea that the users gets what they get in that case.  What would you like them to get for that case?

Reply:
Also, when getting to coding this, I thought it might make just as much sense to just have a separate option like `-days_back` instead of overloading `-run_start_time` this way, but...

Reply:
OH, and I missed that you want a ValueError on a positive float.  My issue with that was I figured there was a whole area of "reasonableness" checking that we are not doing on these run_start_times, but if you want that specific case to get a ValueError that is fine.

Reply:
Most likely user doesn't get what they want with a positive float.  This is what I meant about punting the error downstream in this case.
```
In [5]: DateTime('3').date
Out[5]: '1997:365:23:58:59.816'

In [6]: DateTime(3).date
Out[6]: '1997:365:23:58:59.816'
```

Reply:
Since this is a testing thing, I don't care that much about raising `ValueError` in which case you just remove the `if run_start < 0` test.  But as shown above, user will end up querying Ska for 20 years of data if they put in a positive value.

Reply:
And yes, I saw 3 would give you 1997.  I was somewhat wondering if you wanted to hack this even more so 3 or -3 would give you the same answer.

Reply:
No, because +3 is basically never a useful thing to do.  It will almost certainly break other parts of the code (you can't start from the middle of loads).  That's why ValueError is the correct response.

Reply:
I meant same effective answer (-3 or 3 taking you 3 days before backstop start).